cpp/src/fil/infer.cu

```diff
@@ -238,19 +238,27 @@ struct tree_aggregator_t {
     : tmp_storage(finalize_workspace) {}
 
   __device__ __forceinline__ void accumulate(
-    vec<NITEMS, float> single_tree_prediction, int tree, int num_rows) {
+    vec<NITEMS, float> single_tree_prediction, int tree) {
     acc += single_tree_prediction;
   }
 
-  __device__ __forceinline__ void finalize(float* out, int num_rows,
+  __device__ __forceinline__ void finalize(float* block_out, int num_rows,
                                            int output_stride,
-                                           output_t transform, int num_trees) {
+                                           output_t transform, int num_trees,
+                                           int log2_threads_per_tree) {
+    __syncthreads();
+    auto per_thread = (vec<NITEMS, float>*)tmp_storage;
+    per_thread[threadIdx.x] = acc;
+    __syncthreads();
+    acc = multi_sum<5>(per_thread, 1 << log2_threads_per_tree,
```

Comment:
Does the reduction tree with 5 children per node work the best?

Reply:
That would be my guess based on benchmarking the multi_sum a long time ago with similar parameters